src/ipc-handlers.ts

```diff
@@ -216,6 +217,99 @@ export async function updateSite(
 	return mergeSiteDetailsWithRunningDetails( userData.sites );
 }
 
+export async function connectWpcomSite(
+	event: IpcMainInvokeEvent,
+	site: SyncSite,
+	localSiteId: string,
+	stagingSites: SyncSite[]
+) {
+	const userData = await loadUserData();
+	const currentUserId = userData.authToken?.id;
+
+	if ( ! currentUserId ) {
+		throw new Error( 'User not authenticated' );
+	}
+
+	userData.connectedWpcomSites = userData.connectedWpcomSites || {};
+	userData.connectedWpcomSites[ currentUserId ] =
+		userData.connectedWpcomSites[ currentUserId ] || [];
+
+	const connections = userData.connectedWpcomSites[ currentUserId ];
+
+	const addConnection = ( siteToConnect: SyncSite ) => {
+		const isAlreadyConnected = connections.some(
+			( conn ) => conn.id === siteToConnect.id && conn.localSiteId === localSiteId
+		);
+
+		if ( ! isAlreadyConnected ) {
+			connections.push( { ...siteToConnect, localSiteId } );
+		}
+	};
+
+	addConnection( site );
+
+	// Connect staging sites if any
+	if ( stagingSites && stagingSites.length > 0 ) {
+		stagingSites.forEach( ( stagingSite ) => {
+			addConnection( {
+				...stagingSite,
+				isStaging: true,
+				stagingSiteIds: [],
+			} );
+		} );
+	}
+
+	await saveUserData( userData );
+
+	return connections.filter( ( conn ) => conn.localSiteId === localSiteId );
+}
+
+export async function disconnectWpcomSite(
+	event: IpcMainInvokeEvent,
+	siteIds: number[],
+	localSiteId: string
+) {
+	const userData = await loadUserData();
+	const currentUserId = userData.authToken?.id;
+
+	if ( ! currentUserId ) {
+		throw new Error( 'User not authenticated' );
+	}
+
+	const connections = userData.connectedWpcomSites?.[ currentUserId ] || [];
+
+	const updatedConnections = connections.filter(
+		( conn ) => ! ( siteIds.includes( conn.id ) && conn.localSiteId === localSiteId )
+	);
+
+	userData.connectedWpcomSites = {
+		...userData.connectedWpcomSites,
+		[ currentUserId ]: updatedConnections,
+	};
+
+	await saveUserData( userData );
+
+	return updatedConnections.filter( ( conn ) => conn.localSiteId === localSiteId );
+}
+
+export async function getConnectedWpcomSites(
+	event: IpcMainInvokeEvent,
+	localSiteId: string
+): Promise< SyncSite[] > {
+	const userData = await loadUserData();
+	const currentUserId = userData.authToken?.id;
+
+	if ( ! currentUserId ) {
+		return [];
+	}
+
+	return (
+		userData?.connectedWpcomSites?.[ currentUserId ]?.filter(
```

Comment:
We don't need this question mark. We already accessed this object on line 300.

```suggestion
		userData.connectedWpcomSites?.[ currentUserId ]?.filter(
```

Reply:
Fixed in https://github.com/Automattic/studio/pull/610/commits/28b3e1e9bee7e92e83254d8e5b93c08d75a6b372